pkg/client/didexchange/client.go

```diff
@@ -284,31 +285,33 @@ func (c *Client) GetConnectionAtState(connectionID, stateID string) (*Connection
 	}, nil
 }
 
-// SaveConnection saves a new connection record in completed state and returns the generated connectionID.
-func (c *Client) SaveConnection(req *ConnectionReq) (string, error) {
-	connectionID := uuid.New().String()
-
-	rec := &connection.Record{
-		ConnectionID:    connectionID,
-		State:           connection.StateNameCompleted,
-		ThreadID:        req.ThreadID,
-		ParentThreadID:  req.ParentThreadID,
-		TheirLabel:      req.TheirLabel,
-		TheirDID:        req.TheirDID,
-		MyDID:           req.MyDID,
-		ServiceEndPoint: req.ServiceEndPoint,
-		RecipientKeys:   req.RecipientKeys,
-		RoutingKeys:     req.RoutingKeys,
-		InvitationID:    req.InvitationID,
-		Namespace:       connection.MyNSPrefix,
-	}
-
-	err := c.didexchangeSvc.SaveConnectionRecord(rec)
+// CreateConnection creates a new connection between myDID and theirDID and returns the connectionID.
+func (c *Client) CreateConnection(myDID string, theirDID *did.Doc, options ...ConnectionOption) (string, error) {
```

Comment:
Probably theriDIDDoc would be more helpful to differentiate between DID and DIDDoc as currently we have `myDID`(which is a string) ad `theirDID`(which is of type *did.Doc).